### PR TITLE
High Availability Documentation + Leader/Follower

### PR DESCRIPTION
**Pull request by mbolt35**
Early draft of documentation, how to enable.

### CODE REVIEW

**Comment by mbolt35** [comment]
@dwbrown2 Guessing we also want to try and pull in some of the other internal documentation we have on high-availability strategies?

**Comment by dwbrown2** [comment]
@bstuder99 would you have time to review?

File: high-availability.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+High Availability Kubecost
+==========================
+
+Running kubecost in high availability mode is a feature that relies on multiple Kubecost replica pods implementing the [ETL Bucket Backup](https://raw.githubusercontent.com/kubecost/docs/main/etl-backup.md) feature combined with a Leader/Follower implementation which ensures that there always exists exactly one leader across all replicas.
```
Comment (dwbrown2) [inline high-availability.md:4]: I would add... "Note: HA mode is Available in Kubecost Enterprise today"

File: high-availability.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+The Leader/Follower implementation leverages a `coordination.k8s.io/v1` `Lease` resource to manage the election of a leader when necessary. To control access of the backup from the ETL pipelines, a `RWStorageController` is implemented to ensure the following: 
+* Followers block on all backup reads, and poll bucket storage for any backup reads every 30 seconds.
+* Followers no-op on any backup writes.
+* Followers which receive Queries into a backup store will not stack on pending reads, preventing external queries from blocking.
+* Followers promoted to Leader will drop all locks and receive write privileges.
+* Leaders behave identically to a single Kubecost install. 
+
+![Leader/Follower](https://raw.githubusercontent.com/kubecost/docs/main/images/leader-follower.png)
+
+## How to Configure High Availability
```
Comment (dwbrown2) [inline high-availability.md:17]: I would move this section above architecture.

File: high-availability.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+In order to enable the leader/follower and high availability features, the following must also be configured:
+* Replicas are set to a value greater than 1
+* ETL FileStore is Enabled (enabled by default)
+* [ETL Bucket Backup](https://raw.githubusercontent.com/kubecost/docs/main/etl-backup.md) is Configured
+
+For eexample, using our helm chart, the following is an acceptable configuration:
```
Comment (dwbrown2) [inline high-availability.md:24]: example typo

File: high-availability.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+# Used for HA mode in Business & Enterprise tier
+kubecostDeployment:
+  # Select a number of replicas of Kubecost pods to run 
+  replicas: 5
+  # Enable Leader/Follower Election 
+  leaderFollower:
+    enabled: true
+```
+
+----
```
Comment (dwbrown2) [inline high-availability.md:54]: How to verify backup is properly enabled/working?

**Review by dwbrown2 — COMMENTED** [review]
Awesome to see this!

File: high-availability.md (hunk 4 above)
Comment (mbolt35) [reply high-availability.md:54]: This is part of [ETL Bucket Backup](https://github.com/kubecost/docs/blob/main/etl-backup.md) docs.

File: high-availability.md (hunk 1 above)
Comment (dwbrown2) [reply high-availability.md:4]: It is, have a consistent language that you'd like to use?

Comment (bstuder99) [reply high-availability.md:4]: Not really related to style, I just think we should reword to "...is only available in Kubecost Enterprise." If I wanted to go ahead and add this, would I have to make a separate PR?

Comment (dwbrown2) [reply high-availability.md:4]: @mbolt35 what would you suggest?

@bstuder99 I think that's fine but I also think calling this out more clearly can help more users be aware. Examples:

https://slack.com/help/articles/360000394286-Audit-logs-on-Enterprise-Grid
https://guide.kubecost.com/hc/en-us/articles/4407595985047-User-Management-SSO-SAML
https://guide.kubecost.com/hc/en-us/articles/4407601815191-ETL-S3-Backup

Comment (bstuder99) [reply high-availability.md:4]: Can you elaborate what you mean by "calling this out more clearly"? I feel a bolded Note added at the top of the page in the links you provided is pretty clear and draws the reader's eye.

Comment (dwbrown2) [reply high-availability.md:4]: Yep, that's all I was suggesting and wanted to see what you thought!

Comment (mbolt35) [reply high-availability.md:4]: @bstuder99 I have no problem with you pulling down this branch and modifying it directly!

Comment (bstuder99) [reply high-availability.md:4]: Ok, can you actually explain briefly how I do that? Not super well-versed with modifying PRs in Git.

Comment (bstuder99) [reply high-availability.md]: I've gone ahead and added the disclaimer about Enterprise. If that was the only thing left, is this good to merge?

**Pull request merged**